components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -177,16 +177,23 @@ internal class RustPushConnection(
                 registrationId = token,
                 databasePath = databasePath
             )
-            return true
+            pushApi = api!!
```

Comment:
This is actually the key to the change - we'd enter this block with the new token but never tell the server about it. Every started we'd rinse and repeat - always getting a new token but never telling the server.

Many of the other changes are to be more robust - eg, our subscriptions might vanish and we'd never know.